proxy/config.go

```diff
@@ -99,10 +99,11 @@ type Config struct {
 	// Cache settings
 	// --
 
-	CacheEnabled   bool   // cache status
-	CacheSizeBytes int    // Cache size (in bytes). Default: 64k
-	CacheMinTTL    uint32 // Minimum TTL for DNS entries (in seconds).
-	CacheMaxTTL    uint32 // Maximum TTL for DNS entries (in seconds).
+	CacheEnabled    bool   // cache status
+	CacheSizeBytes  int    // Cache size (in bytes). Default: 64k
+	CacheMinTTL     uint32 // Minimum TTL for DNS entries (in seconds).
+	CacheMaxTTL     uint32 // Maximum TTL for DNS entries (in seconds).
+	CacheOptimistic bool   // Optimistic cache status
```

Comment:
It's probably better to place this one right after `CacheEnabled` to get a better struct alignment.

Reply:
fixed